### PR TITLE
+.json, +user transforms, +base transforms

### PR DESCRIPTION
**Pull request by NumminorihSF**
* some readme.md update
* translator now use 1 regexp pattern
* translator can works with user RegExp pattern
* translator can work without translate (wow)
* can use user's pattern separator
* can use user's transforms
* can use .json files
* add capitalize transform
* add capitalizeEvery transform
* add reverse transform

### CODE REVIEW

File: lib/translator.js [hunk 1]
```diff
@@ -1,80 +1,167 @@
...
+    return content.toLowerCase().replace(/[^\s\.!\?]+(.)/g, function(word){
+      return this.capitalize(word);
+    });
+  },
+
+  reverse: function(content){
+    if (!content) {
+      return new Error('No content to transform.');
+    }
+    var res = '';
```
Comment (arathunku) [inline lib/translator.js]: What do you think of using reduce here in a following way:

```javascript
return content.split('').reduce(function(r, c) { return c+r;}, '')
```
? 

Sorry for nitpicking

File: lib/translator.js [hunk 2]
```diff
@@ -1,80 +1,167 @@
...
+          return baseTransforms[transformName](res, self.dictionary);
+        } else {
+          return new Error('No such transform: '+ transformName);
+        }
+      }, value);
+
+      if (res instanceof Error) {
+        return resultPromise.reject(res);
+      }
+     // console.log('c', content);
```
Comment (arathunku) [inline lib/translator.js]: Probably to remove

File: package.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
     "name": "gulp-translator",
     "main": "translator-gulp.js",
-    "version": "0.1.0",
-    "description": "Translate your files by using {{ syntax }} and locales/something.yml files.",
+    "version": "0.1.1",
```
Comment (arathunku) [inline package.json]: :+1: :+1: :+1:

**Comment by arathunku** [comment]
@NumminorihSF thank for PR <3 <3 

I'm happy to merge it and release on npm but it would be good to add some tests for the changes and fixes added here. 

PS. I'll fix houndci, sorry for redundant comments from it!

File: lib/translator.js (hunk 1 above)
Comment (NumminorihSF) [inline lib/translator.js]: Sounds good. It works with `.reduceRight()` at updated PR

File: lib/translator.js (hunk 2 above)
Comment (NumminorihSF) [inline lib/translator.js]: Should i remove comments? Or something else? Can\'t understand GitHub line lines. Does it show commented line as the last in code window?

File: test/translator-gulp.js [hunk 4]
```diff
@@ -163,6 +188,98 @@ describe('gulp-translator', function() {
...
+
+
+
+
+
+
+
+
+
+
```
Comment (arathunku) [inline test/translator-gulp.js]: Lots of empty lines

Comment (NumminorihSF) [inline test/translator-gulp.js]: ouch, sorry. just forgive to delete it

**Comment by arathunku** [comment]
@NumminorihSF thanks!

**Pull request merged**